tools/topology/sof/pipe-smart-amplifier-playback.m4

```diff
@@ -35,16 +35,26 @@ ifdef(`SMART_UUID',`', `define(`SMART_UUID', smart_amp_comp_uuid)');
 # Controls
 #
 
+ifdef(`SMART_AMPS_NUM',`', `define(`SMART_AMPS_NUM',`4')')
 # initial config params for smart_amp, aligned with struct sof_smart_amp_config
-CONTROLBYTES_PRIV(SMART_AMP_priv,
+ifelse(SMART_AMPS_NUM, `4',
+`CONTROLBYTES_PRIV(SMART_AMP_priv,
+`		bytes "0x53,0x4f,0x46,0x00,0x00,0x00,0x00,0x00,'
+`		0x18,0x00,0x00,0x00,0x00,0x00,0x00,0x03,'
+`		0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,'
+`		0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,'
+`		0x18,0x00,0x00,0x00,0x08,0x00,0x00,0x00,'
+`		0x00,0x01,0xff,0xff,0xff,0xff,0xff,0xff,'
+`		0xff,0xff,0x00,0x01,0xff,0xff,0xff,0xff"')',
+`ifelse(SMART_AMPS_NUM, `2',
+`CONTROLBYTES_PRIV(SMART_AMP_priv,
 `		bytes "0x53,0x4f,0x46,0x00,0x00,0x00,0x00,0x00,'
 `		0x18,0x00,0x00,0x00,0x00,0x00,0x00,0x03,'
 `		0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,'
 `		0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,'
 `		0x18,0x00,0x00,0x00,0x04,0x00,0x00,0x00,'
 `		0x00,0x01,0xff,0xff,0xff,0xff,0xff,0xff,'
-`		0x00,0x01,0x02,0x03,0xff,0xff,0xff,0xff"'
-)
+`		0x00,0x01,0x02,0x03,0xff,0xff,0xff,0xff"')', `')')
```

Comment:
may be better to reuse the SMART_FB_CHANNELS macro.
```
ifelse(SMART_FB_CHANNELS, `8',
`define(`FB_CHMAP' `0xff,0xff,0x00,0x01,0xff,0xff,0xff,0xff')',
`define(`FB_CHMAP' `0x00,0x01,0x02,0x03,0xff,0xff,0xff,0xff')'
)

CONTROLBYTES_PRIV(SMART_AMP_priv,
`		bytes "0x53,0x4f,0x46,0x00,0x00,0x00,0x00,0x00,'
`		0x18,0x00,0x00,0x00,0x00,0x00,0x00,0x03,'
`		0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,'
`		0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,'
`		0x18,0x00,0x00,0x00,0x04,0x00,0x00,0x00,'
`		0x00,0x01,0xff,0xff,0xff,0xff,0xff,0xff,'
`		FB_CHMAP"')'

```

Reply:
Hi Keyon,
Thanks for your quick review.
there are two variables changing here in smart_Amp_config structure.
uint32_t feedback_channels & feedback_ch_map
if there are 2 smart amps the feedback_channel=4 & 0x00,0x01,0x02,0x03,0xff,0xff,0xff,0xff
for 4 smart_amps the feedback channel=8 & [0xff,0xff,0x00,0x01,0xff,0xff,0xff,0xf]->doesn't sound right as we are actually giving only 2 channel feedback, but i believe this is for the smart_amp_test we have today.
Also the current sof-tgl-max98373_rt5682 file has this SMART_FB_CHANNEL defined as 8.
So will it be right to use it this way? Should SMART_FB_CHANNEL be changed to 4 in the sof-tgl-max98373-rt5682.m4 as well? will it have any other implication/side effects because of this?

Reply:
@ryans-lee fyi - any comments ?

Reply:
@keyonjie : I did try some basic testing with the changes you suggested and the change of FB_CHANNELS to 4 in sof-tgl-max98373-rt5682.m4 doesnt seem to have any issues. Hence updated the patch with your suggestion.